front/src/components/member/MemberUpdateModal.vue

```diff
@@ -35,18 +38,33 @@
 <script>
 import api from "@/api/member";
 
+const NICKNAME_REGEX = "^[a-zA-Zㄱ-ㅎㅏ-ㅣ가-힣0-9]+$";
+const VIOLATE_REGEX = (val) => !val.match(NICKNAME_REGEX);
+const IS_VALUE_SAME = (newVal, oldVal) => newVal === oldVal;
+
 export default {
-  name: "PostCreateModal",
+  name: "MemberUpdateModal",
   data() {
     return {
       rendered: false,
       newNickname: "",
+      rules: {
+        regexViolation: (newVal) =>
+          !VIOLATE_REGEX(newVal) ||
+          "닉네임은 숫자/한글/영어로 이루어져야 합니다.",
+      },
     };
   },
   computed: {
     member() {
       return this.$store.getters["member/getMember"];
     },
+    isUpdateDisabled() {
+      return (
+        IS_VALUE_SAME(this.member.nickname, this.newNickname) ||
+        VIOLATE_REGEX(this.newNickname)
+      );
```

Comment:
```suggestion
      return (
        (this.member.nickname === this.newNickname) ||
        VIOLATE_REGEX(this.newNickname)
      );
```

따로 rules에서 관리를 안할 로직이라면 빼둘 필요가 있을까 싶네요.